nomad/structs/variables_test.go

```diff
@@ -8,9 +8,177 @@ import (
+				Path:        "custom/test/path",
+				Lock:        nil,
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 200,
+				ModifyTime:  1687251885,
+			},
+			expectedOutput: false,
+		},
+
+		{
+			name: "lock equal",
+			inputMetadata: VariableMetadata{
+				Namespace: "default",
+				Path:      "custom/test/path",
+				Lock: &VariableLock{
+					ID:        "896bdbef-8ce7-4b1d-9b4c-4e6c5639196d",
+					TTL:       20 * time.Second,
+					LockDelay: 5 * time.Second,
+				},
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251868,
+			},
+			inputMetadataFn: VariableMetadata{
+				Namespace: "default",
+				Path:      "custom/test/path",
+				Lock: &VariableLock{
+					ID:        "896bdbef-8ce7-4b1d-9b4c-4e6c5639196d",
+					TTL:       20 * time.Second,
+					LockDelay: 5 * time.Second,
+				},
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251868,
+			},
+			expectedOutput: true,
+		},
+		{
+			name: "lock unequal",
+			inputMetadata: VariableMetadata{
+				Namespace: "default",
+				Path:      "custom/test/path",
+				Lock: &VariableLock{
+					ID:        "896bdbef-8ce7-4b1d-9b4c-4e6c5639196d",
+					TTL:       20 * time.Second,
+					LockDelay: 5 * time.Second,
+				},
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251876,
+			},
+			inputMetadataFn: VariableMetadata{
+				Namespace: "default",
+				Path:      "custom/test/path",
+				Lock: &VariableLock{
+					ID:        "896bdbef-8ce7-4b1d-9b4c-4e6c5639196d",
+					TTL:       20 * time.Second,
+					LockDelay: 15 * time.Second,
+				},
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251876,
+			},
+			expectedOutput: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			must.Eq(t, tc.expectedOutput, tc.inputMetadata.Equal(tc.inputMetadataFn))
+		})
+	}
+}
+
+func TestVariableMetadata_Copy(t *testing.T) {
+	ci.Parallel(t)
+
+	testCases := []struct {
+		name                  string
+		inputVariableMetadata *VariableMetadata
+	}{
+		{
+			name: "no lock",
+			inputVariableMetadata: &VariableMetadata{
+				Namespace:   "default",
+				Path:        "custom/test/path",
+				Lock:        nil,
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251876,
+			},
+		},
+		{
+			name: "lock",
+			inputVariableMetadata: &VariableMetadata{
+				Namespace: "default",
+				Path:      "custom/test/path",
+				Lock: &VariableLock{
+					ID:        "896bdbef-8ce7-4b1d-9b4c-4e6c5639196d",
+					TTL:       20 * time.Second,
+					LockDelay: 15 * time.Second,
+				},
+				CreateIndex: 10,
+				CreateTime:  1687251815,
+				ModifyIndex: 100,
+				ModifyTime:  1687251876,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualOutput := tc.inputVariableMetadata.Copy()
```

Comment:
We're missing a `Copy` of the `VariableLock`, so while these will pass equality, both `Lock` fields are pointing to the exact same object and so this test doesn't fully exercise the `Equal` check.

Reply:
Oh yes, thanks for catching this. I have forced pushed a change to resolve this and wonder if I should add checks to other copy functions that might be missing this.